soc/xtensa/intel_adsp/tools/cavstool.py

```diff
@@ -329,6 +484,45 @@ def ipc_command(data, ext_data):
         ext_data = t - ipc_timestamp
         ipc_timestamp = t
         send_msg = True
+    elif data == 5: # HDA RESET (init if not exists)
+        stream_id = ext_data & 0xff
+        if stream_id in hda_streams:
+            hda_streams[stream_id].reset()
+        else:
+            hda_str = HDAStream(stream_id)
+            hda_streams[stream_id] = hda_str
+    elif data == 6: # HDA CONFIG
+        stream_id = ext_data & 0xFF
+        buf_len = ext_data >> 8 & 0xFFFF
+        hda_str = hda_streams[stream_id]
+        hda_str.config(buf_len)
+    elif data == 7: # HDA START
+        stream_id = ext_data & 0xFF
+        hda_streams[stream_id].start()
+
+    elif data == 8: # HDA STOP
+        stream_id = ext_data & 0xFF
+        hda_streams[stream_id].stop()
+    elif data == 9: # HDA VALIDATE
+        stream_id = ext_data & 0xFF
+        hda_str = hda_streams[stream_id]
+        hda_str.debug()
+        is_ramp_data = True
```

Comment:
OK, you got me: I can't figure out what "ramp" means here. :)  Maybe just call this flag "valid" or something?

Also: maybe this is too much intelligence for the script?  If you just had a "read this offset from a stream buffer" command, you could do the validation in the C code where the buffer is generated, which is slower but IMHO a lot clearer and more flexible.